### PR TITLE
Thibault/importing data2

### PR DESCRIPTION
**Pull request by thibmonsel**
This PR holds several new features, and modifications : 

1/ Took out the Matplotlib library since I found a better package for plotting in c++ (https://alandefreitas.github.io/matplotplusplus/)
2/Dockerfile changed.
3/ Took out the << operator for unordered_map
4/ Created 2 plot functions : 
-  plot the data from an AlternativePerformance object
- plot the profile from a Profiles object
5/ Tested ProfInitializer on a toy dataset and 2 datasets taken from Olivier's PyMCDA.
6/Fixed loadDataset function from DatasetGenerator class.
7/ Added new function in DataGenerator that checks that the number of categories given in the <categories> tag of the xml dataset file has at least one candidate for each category.

In the TestProfileInitializer there is a lot of commenting since it is the code to reproduce the changeMode() bug in our code.

### CODE REVIEW

File: include/utils.h [hunk 1]
```diff
@@ -168,4 +159,140 @@ inline std::vector<Perf> createVectorPerfWithNoPerf(std::string id,
...
+  }
+  return values;
+}
+
+/**
+ * Check if a key is present in a unordered map
+ * @param map unordered map of <int,int>
+ *
+ */
+inline bool checkKey(std::unordered_map<int, int> &map, int key) {
```
Comment (Mostah) [inline include/utils.h:263]: I'm wondering why adding a function that does just this line `map.find(key) != map.end()` ?

File: test/learning/TestInitializeProfile.cpp [hunk 2]
```diff
@@ -176,7 +164,42 @@ TEST(TestProfileInitializer, TestInitializeProfiles) {
...
+
+  // in3, in4 don't work since it doesnt have alternatives present for
+  // all of its categories | in1 works, in7 works but takes long time
+  // for in7, cat1 is people that are likely to die or develop an extreme
+  // condition of a disease
+  AlternativesPerformance ap = data.loadDataset(filename);
+  int nbCat = data.getNumberOfCategories(filename);
+  int nbCrit = data.getNumberOfCriteria(filename);
+  MRSortModel model = MRSortModel(nbCat, nbCrit);
+  // When we initialize the profile change mode stops working. We cant change
```
Comment (Mostah) [inline test/learning/TestInitializeProfile.cpp]: is this comment still accurate?

File: test/types/TestDataGenerator.cpp [hunk 3]
```diff
@@ -572,3 +586,62 @@ TEST(TestDataGenerator, TestGetCriterionCategoryLimitsModelFakeCritId) {
     FAIL() << "should have throw invalid_argument error.";
   }
 }
+
+TEST(TestDataGenerator, TestImportedDataset) {
+  // IMPORTANT NOTE : cannot have any xml tag with only digits ie <1> c++ cant
```
Comment (Mostah) [inline test/types/TestDataGenerator.cpp]: There is no test in both `TEST` here

File: test/types/TestDataGenerator.cpp [hunk 4]
```diff
@@ -572,3 +586,62 @@ TEST(TestDataGenerator, TestGetCriterionCategoryLimitsModelFakeCritId) {
...
+        err.what(),
+        std::string(
+            "The number of categories found in the dataset does not match the "
+            "number of categories given in the xml <categories> tag."));
+  } catch (...) {
+    FAIL() << "should have throw invalid_argument error.";
+  }
+}
+
+TEST(TestDataGenerator, TestDataCompatabilityRealDatasetWork) {
```
Comment (Mostah) [inline test/types/TestDataGenerator.cpp:624]: same

**Review by Mostah — APPROVED** [review]
Just a few things, otherwise it LGTM

File: include/utils.h (hunk 1 above)
Comment (thibmonsel) [reply include/utils.h:263]: for readability purpose but for sure it could be put inside the function itself

**Pull request merged**